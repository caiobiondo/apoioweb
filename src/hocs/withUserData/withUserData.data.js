import gql from 'graphql-tag';

export const UserDataQuery = gql`
  query UserDataQuery {
    user {
      codigo
      cdCanalCaptacao
      codigoCentro
      nomeCompleto
      cdPapelAtivo
      acl {
        stock
        customers
        orders
        myScore
        training
        magazine
        cnd
        careerPlan
      }
      estrutura {
        codigo
        codigoTipo
        ciclo {
          numero
        }
        gerenciaMercado {
          codigo
        }
        regiaoEstrategica {
          codigo
        }
        gerenciaVenda {
          codigo
        }
        setor {
          codigo
        }
      }
    }
  }
`;
