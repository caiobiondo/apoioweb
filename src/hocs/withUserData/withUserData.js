import React, { PureComponent } from 'react';
import { graphql } from 'react-apollo';
import { UserDataQuery } from './withUserData.data';
import { Loading } from 'natura-ui';
import LocalStorageData from 'infra/LocalStorageData';
import { get } from 'lodash';

export const WithUserData = Component => {
  return class WithUserDataComponent extends PureComponent {
    render() {
      const { data } = this.props; //const data = userMock;
      if (!data || !data.user) {
        return <Loading background="transparent" />;
      }
      // workaround: sellers api does not return user's current cycle
      // this workaround uses localStorage's defined by webfv
      const { cycle } = new LocalStorageData();
      const user = {
        ...data.user,
        estrutura: {
          ...data.user.estrutura,
          ciclo: [
            {
              numero: get(data.user, 'estrutura.ciclo[0].numero', null) || cycle,
            },
          ],
        },
      };
      return <Component {...this.props} user={user} />;
    }
  };
};

export default function withUserData(Component, graphqlConnect = graphql) {
  const WithUserDataComponent = WithUserData(Component);

  return graphqlConnect(UserDataQuery)(WithUserDataComponent);
}
// prettier-ignore
const userMock = {  
  "variables":{  },
  "loading":false,
  "networkStatus":7,
  "user":{  
     "codigo":16578,
     "cdCanalCaptacao":8,
     "codigoCentro":5800,
     "nomeCompleto":"NADIRCE IZIDIO DE OLIVEIRA",
     "cdPapelAtivo":2,
     "acl":{  
        "stock":true,
        "customers":true,
        "orders":true,
        "myScore":true,
        "training":true,
        "magazine":true,
        "cnd":true,
        "careerPlan":null,
        "__typename":"AclStructure"
     },
     "estrutura":{  
        "codigo":10497,
        "codigoTipo":5,
        "ciclo":[  
           {  
              "numero":201902,
              "__typename":"UserCycle"
           }
        ],
        "gerenciaMercado":{  
           "codigo":2,
           "__typename":"MarketManagement"
        },
        "regiaoEstrategica":{  
           "codigo":2,
           "__typename":"StrategicRegion"
        },
        "gerenciaVenda":{  
           "codigo":145,
           "__typename":"SalesManagement"
        },
        "setor":{  
           "codigo":176,
           "__typename":"Sector"
        },
        "__typename":"UserStructure"
     },
     "__typename":"User"
  }
};
