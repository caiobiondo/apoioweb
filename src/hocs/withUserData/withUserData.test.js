import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { WithUserData } from './withUserData';

class WrappedComponent extends React.Component {
  render() {
    return <div />;
  }
}

describe('withUserData HOC', () => {
  it('should render the child component with user props', () => {
    const renderer = new ShallowRenderer();
    const data = {
      user: {
        codigo: 11,
        cdCanalCaptacao: 12,
        codigoCentro: 13,
      },
    };
    const Component = WithUserData(WrappedComponent);

    renderer.render(<Component data={data} />);
    const result = renderer.getRenderOutput();

    expect(result).toMatchSnapshot();
  });

  it('should render a loading indicator', () => {
    const renderer = new ShallowRenderer();
    const data = null;
    const Component = WithUserData(WrappedComponent);

    renderer.render(<Component data={data} />);
    const result = renderer.getRenderOutput();

    expect(result).toMatchSnapshot();
  });

  it('should render a loading indicator', () => {
    const renderer = new ShallowRenderer();
    const data = {
      user: null,
    };
    const Component = WithUserData(WrappedComponent);

    renderer.render(<Component data={data} />);
    const result = renderer.getRenderOutput();

    expect(result).toMatchSnapshot();
  });
});
