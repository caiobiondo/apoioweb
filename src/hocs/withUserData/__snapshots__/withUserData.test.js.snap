// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withUserData HOC should render a loading indicator 1`] = `
<Loading
  background="transparent"
  visible={true}
  zIndex={9999}
/>
`;

exports[`withUserData HOC should render a loading indicator 2`] = `
<Loading
  background="transparent"
  visible={true}
  zIndex={9999}
/>
`;

exports[`withUserData HOC should render the child component with user props 1`] = `
<WrappedComponent
  data={
    Object {
      "user": Object {
        "cdCanalCaptacao": 12,
        "codigo": 11,
        "codigoCentro": 13,
      },
    }
  }
  user={
    Object {
      "cdCanalCaptacao": 12,
      "codigo": 11,
      "codigoCentro": 13,
    }
  }
/>
`;
